File: mac/PabloCompanion/Models/PracticeTopic.swift
import Foundation

/// A practice session topic from the backend catalog.
struct PracticeTopic: Codable, Identifiable, Sendable {
    let id: String
    let name: String
    let description: String
    let category: String
    let estimatedDurationMinutes: Int

    enum CodingKeys: String, CodingKey {
        case id, name, description, category
        case estimatedDurationMinutes = "estimated_duration_minutes"
    }
}

struct PracticeTopicListResponse: Codable, Sendable {
    let data: [PracticeTopic]
    let total: Int
}

File: windows/PabloCompanion/Models/PracticeTopic.cs
namespace PabloCompanion.Models;

/// <summary>
/// A practice session topic from the backend catalog.
/// </summary>
public sealed record PracticeTopic(
    string Id,
    string Name,
    string Description,
    string Category,
    int EstimatedDurationMinutes
);

public sealed record PracticeTopicListResponse(
    PracticeTopic[] Data,
    int Total
);
